### ISSUE TITLE
Errors occered while running utils/validate_data_dir.sh when I used CSJ corpus

### ISSUE THREAD

**Issue by muramasa2** [kaldi-asr/kaldi#4126]
Hi, I am using egs/csj/s5 to train model, when comes to command steps/make_mfcc.sh --nj 50 --cmd run.pl data/train exp/make_mfcc/train mfcc
steps/make_mfcc.sh: moving data/train/feats.scp to data/train/.backup
utils/validate_data_dir.sh, the job failed, and the LOG shows:

```
steps/make_mfcc.sh --nj 50 --cmd run.pl data/train exp/make_mfcc/train mfcc
steps/make_mfcc.sh: moving data/train/feats.scp to data/train/.backup
utils/validate_data_dir.sh: text contains 58815 lines with non-printable characters
```

The text file read in the script(utils/validate_data_dir.sh) is exist and it looks fine, but it doesn't work.
And I try to roll back to previous commit before than the commit 31c2baeb53737f11f9a06a7ca9cc27d9b0e6438a.

After that it became able to work, so i think the cause of this error is this commit has bugs. 
Anyone have any idea about the cause?
Thank you!

**Comment by kkm000** [comment]
Thanks, that's the culprit. The check added in #4094 won't work with the C locale that we normally use.

**Comment by alex-ht** [comment]
Hi, I just made a PR #4130.

**Comment by danpovey** [comment]
@kkm000 you can merge this and similar things... please remember to always do 'sqash and merge' (keeps commit history clean); and I generally edit the commit message for readability and to add certain standard tags, and remove the "detailed commit message" if I feel it isn't helpful.  In this case tag would be [build], I think.

**Comment by kkm000** [comment]
Fixed in #4130 

@danpovey, sure.
